### PR TITLE
Fix redirects and dependencies of two projects

### PR DESCRIPTION
**Pull request by mattwar**
Fix for https://github.com/dotnet/roslyn/issues/14090

This change fixes the assembly redirects and dependencies of two projects that are causing build failures due to parallel build race conditions.  In many cases the redirects have been removed because the project does not actually depend on them, so they may not be present in the output directory at the time the redirects are processed. Also, in some cases, actual dependencies were not present and the projects were just getting lucky with parallel build order and relying on the fact that all projects were outputting to the same directory.

@jasonmalinowski @jaredpar @CyrusNajmabadi  please take a look.

### CODE REVIEW

**Comment by tmat** [comment]
@agocke Re facades. I wonder if instead of removing the attributes we need to add dependency on the facades.

File: src/VisualStudio/VisualStudioInteractiveComponents/VisualStudioInteractiveComponents.csproj [hunk 1]
```diff
@@ -58,6 +58,10 @@
       <IncludeOutputGroupsInVSIXLocalOnly>DebugSymbolsProjectOutputGroup</IncludeOutputGroupsInVSIXLocalOnly>
       <ForceIncludeInVSIX>true</ForceIncludeInVSIX>
     </ProjectReference>
+    <ProjectReference Include="..\..\Interactive\EditorFeatures\VisualBasic\BasicInteractiveEditorFeatures.vbproj"> 
+        <Project>{849e516a-595f-474b-adb4-e099f921cedf}</Project>
+        <Name>BasicInteractiveEditorFeatures</Name>
```
Comment (jasonmalinowski) [inline src/VisualStudio/VisualStudioInteractiveComponents/VisualStudioInteractiveComponents.csproj]: Why this change? I believe we didn't have this or else we were bundling VB interactive which we don't support?

Comment (mattwar) [inline src/VisualStudio/VisualStudioInteractiveComponents/VisualStudioInteractiveComponents.csproj]: It was being packaged at some point, and failure to do so when I put this project into a sub directory.

Comment (mattwar) [inline src/VisualStudio/VisualStudioInteractiveComponents/VisualStudioInteractiveComponents.csproj]: Strangely, even without VB interactive in the project dependency, we end up dependent on vb compiler assemblies, and those being packaged too.
